backend/main/models/vote.go

```diff
@@ -308,3 +328,145 @@ func DoesNFTExist(db *s.Database, v *VoteWithBalance) (bool, error) {
 
 	return true, nil
 }
+
+func createVote(db *s.Database, v *Vote) error {
+	// Create Vote
+	err := db.Conn.QueryRow(db.Context,
+		`
+			INSERT INTO votes(proposal_id, addr, choice, composite_signatures, cid, message)
+			VALUES($1, $2, $3, $4, $5, $6)
+			RETURNING id, created_at
+		`, v.Proposal_id, v.Addr, v.Choice, v.Composite_signatures, v.Cid, v.Message).Scan(&v.ID, &v.Created_at)
+
+	return err
+}
+
+func getProposal(db *s.Database, proposalId int) (Proposal, error) {
+	var proposal Proposal
+	err := pgxscan.Get(db.Context, db.Conn, &proposal,
+		`SELECT start_time, community_id from proposals
+		WHERE id = $1`,
+		proposalId)
+
+	return proposal, err
+}
+
+func checkForEarlyVoteAchievement(db *s.Database, v *Vote, p Proposal) error {
```

Comment:
it's not clear from the function name that this function actually does an `INSERT` and updates the DB. This function is doing two things, checking `isEarlyVote` then it's also updating the DB.  

I think we should put the sql part into a separate function called `updateUserAchievments` then, at the controller level conditionally call that if isEarlyVote is true, which would be returned from this function.